config/config.go

```diff
@@ -772,3 +788,39 @@ const (
 	OOMActionCancel = "cancel"
 	OOMActionLog    = "log"
 )
+
+type autoRandom struct {
+	originAllowAutoRandom bool
+	originAlterPrimaryKey bool
+}
+
+// AutoRandom provide methods related to allow-auto-random.
+var AutoRandom autoRandom
+
+// IsAllow indicates whether `auto_random` is available.
+func (a *autoRandom) IsAllow() bool {
+	return GetGlobalConfig().Experimental.AllowAutoRandom
+}
+
+// SetAllow sets allow-auto-random to `allow`.
+func (a *autoRandom) SetAllow(allow bool) {
+	GetGlobalConfig().Experimental.AllowAutoRandom = allow
+}
+
+// SetupTestConfig set alter-primary-key to false, and set allow-auto-random to true and save their origin values.
+// This method should only be used for the tests in SerialSuite.
+func (a *autoRandom) SetupTestConfig() {
```

Comment:
Can we move this code to the test file if only used in test?

Reply:
The functions in tests are not allowed to be exported. It is impossible to move them to different tests unless we make a few copies.

It seems that we are in a dilemma between "Single Responsibility" and "DRY".

Reply:
Could we put it to `util/testutil/testutil.go`?

Reply:
@zimulala Thanks. Updated, PTAL